app/models/manageiq/providers/microsoft/infra_manager.rb

```diff
@@ -37,12 +37,37 @@ def self.auth_url(hostname, port = nil)
     URI::HTTP.build(:host => hostname, :port => port || 5985, :path => "/wsman").to_s
   end
 
+  def self.build_connect_params(options)
+    connect_params  = {
+      :user         => options[:user],
+      :password     => MiqPassword.try_decrypt(options[:password]),
+      :endpoint     => options[:endpoint] || auth_url(options[:hostname], options[:port]),
+      :disable_sspi => true
+    }
+
+    if options[:security_protocol] == "kerberos"
+      connect_params.merge!(
+        :realm           => options[:realm],
+        :basic_auth_only => false,
+        :disable_sspi    => false
+      )
+    end
+
+    connect_params
+  end
+
   def connect(options = {})
     raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
     hostname           = options[:hostname] || self.hostname
     options[:endpoint] = self.class.auth_url(hostname, port)
-    connect_params     = build_connect_params(options)
+    options[:user]   ||= authentication_userid(options[:auth_type])
+    options[:password] = MiqPassword.try_decrypt(options[:password]) || authentication_password(options[:auth_type])
```

Comment:
@agrare - you are calling `MiqPassword.try_decrypt...` twice.

Reply:
Good catch

Reply:
Three times if you include raw_connect (it was already doing it twice).
I'll change it to just decrypt in raw_connect.

Reply:
Fixed, tested credential validation and refresh and both work